### PR TITLE
[SPARK-18461][DOCS][StructuredStreaming] Added more information about monitoring streaming queries

### PR DESCRIPTION
**Pull request by tdas**
## What changes were proposed in this pull request?
screen shot 2016-11-15 at 6 27 32 pm
screen shot 2016-11-15 at 6 27 45 pm

### CODE REVIEW

**Comment by tdas** [comment]
@zsxwing @marmbrus

**Comment by marmbrus** [comment]
Thanks, merging to master and 2.1.

File: docs/structured-streaming-programming-guide.md [hunk 1]
```diff
@@ -1087,9 +1087,185 @@ spark.streams().awaitAnyTermination()  # block until any one of them terminates
...
+*/
+{% endhighlight %}
+
+</div>
+<div data-lang="java"  markdown="1">
+
+{% highlight java %}
+StreamingQuery query = ...
+
+System.out.println(query.status);
```
Comment (zsxwing) [inline docs/structured-streaming-programming-guide.md]: nit: `query.status()`

File: docs/structured-streaming-programming-guide.md [hunk 2]
```diff
@@ -1087,9 +1087,185 @@ spark.streams().awaitAnyTermination()  # block until any one of them terminates
...
+*/
+{% endhighlight %}
+
+</div>
+<div data-lang="python"  markdown="1">
+
+{% highlight python %}
+query = ...  // a StreamingQuery
+
+print(query.status)
```
Comment (zsxwing) [inline docs/structured-streaming-programming-guide.md]: there is no `status` method in Python now... We need to add it.

File: docs/structured-streaming-programming-guide.md [hunk 3]
```diff
@@ -1087,9 +1087,185 @@ spark.streams().awaitAnyTermination()  # block until any one of them terminates
...
+</div>
+<div data-lang="java"  markdown="1">
+
+{% highlight java %}
+SparkSession spark = ...
+
+spark.streams.addListener(new StreamingQueryListener() {
+
+    @Overrides void onQueryStarted(QueryStartedEvent queryStarted) {
+        System.out.println("Query started: " + queryTerminated.queryStatus.name);
```
Comment (zsxwing) [inline docs/structured-streaming-programming-guide.md]: nit: `queryTerminated.queryStatus.name` -> `queryTerminated.queryStatus().name()`

File: docs/structured-streaming-programming-guide.md [hunk 4]
```diff
@@ -1087,9 +1087,185 @@ spark.streams().awaitAnyTermination()  # block until any one of them terminates
...
+    }
+    @Overrides void onQueryTerminated(QueryTerminatedEvent queryTerminated) {
+        System.out.println("Query terminated: " + queryTerminated.queryStatus.name);
```
Comment (zsxwing) [inline docs/structured-streaming-programming-guide.md]: nit: queryTerminated.queryStatus.name -> queryTerminated.queryStatus().name()

File: docs/structured-streaming-programming-guide.md [hunk 5]
```diff
@@ -1087,9 +1087,185 @@ spark.streams().awaitAnyTermination()  # block until any one of them terminates
...
+    }
+    @Overrides void onQueryProgress(QueryProgressEvent queryProgress) {
+        System.out.println("Query made progress: " + queryProgress.queryStatus);
```
Comment (zsxwing) [inline docs/structured-streaming-programming-guide.md]: nit: queryTerminated.queryStatus.name -> queryTerminated.queryStatus().name()

File: docs/structured-streaming-programming-guide.md [hunk 6]
```diff
@@ -1087,9 +1087,185 @@ spark.streams().awaitAnyTermination()  # block until any one of them terminates
...
+
+</div>
+<div data-lang="java"  markdown="1">
+
+{% highlight java %}
+SparkSession spark = ...
+
+spark.streams.addListener(new StreamingQueryListener() {
+
+    @Overrides void onQueryStarted(QueryStartedEvent queryStarted) {
```
Comment (zsxwing) [inline docs/structured-streaming-programming-guide.md]: nit: `@ Overrides` -> `@Override`

File: docs/structured-streaming-programming-guide.md [hunk 7]
```diff
@@ -1087,9 +1087,185 @@ spark.streams().awaitAnyTermination()  # block until any one of them terminates
...
+        System.out.println("Query started: " + queryTerminated.queryStatus.name);
+    }
+    @Overrides void onQueryTerminated(QueryTerminatedEvent queryTerminated) {
```
Comment (zsxwing) [inline docs/structured-streaming-programming-guide.md]: nit: `@ Overrides` -> `@Override`

File: docs/structured-streaming-programming-guide.md [hunk 8]
```diff
@@ -1087,9 +1087,185 @@ spark.streams().awaitAnyTermination()  # block until any one of them terminates
...
+        System.out.println("Query terminated: " + queryTerminated.queryStatus.name);
+    }
+    @Overrides void onQueryProgress(QueryProgressEvent queryProgress) {
```
Comment (zsxwing) [inline docs/structured-streaming-programming-guide.md]: nit: `@ Overrides` -> `@Override`

File: docs/structured-streaming-programming-guide.md [hunk 9]
```diff
@@ -1087,9 +1087,185 @@ spark.streams().awaitAnyTermination()  # block until any one of them terminates
...
+})
+{% endhighlight %}
+
+</div>
+<div data-lang="java"  markdown="1">
+
+{% highlight java %}
+SparkSession spark = ...
+
+spark.streams.addListener(new StreamingQueryListener() {
```
Comment (zsxwing) [inline docs/structured-streaming-programming-guide.md]: nit: `spark.streams` -> `spark.streams()`

**Comment by tdas** [comment]
I am making a PR to add status to python. I will fix these doc stuff in that.

**Pull request closed without merging**